src/firefly/js/visualize/saga/ChartsSync.js

```diff
@@ -0,0 +1,54 @@
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ */
+import {take} from 'redux-saga/effects';
+
+import {has} from 'lodash';
+
+import {flux} from '../../Firefly.js';
+import * as TablesCntlr from '../../tables/TablesCntlr.js';
+import * as TableStatsCntlr from '../TableStatsCntlr.js';
+import * as TableUtil from '../../tables/TableUtil.js';
+
+import * as XYPlotCntlr from '../XYPlotCntlr.js';
+import * as HistogramCntlr from '../HistogramCntlr.js';
+
+
+
+/**
+ * this saga handles chart related side effects
+ */
+export function* syncCharts() {
+    var tableStatsState, xyPlotState, histogramState;
+
+    while (true) {
+        const action= yield take([TableStatsCntlr.SETUP_TBL_TRACKING, TablesCntlr.TABLE_NEW_LOADED]);
+        const request= action.payload.request;
+        switch (action.type) {
+            case TableStatsCntlr.SETUP_TBL_TRACKING:
+                const {tblId} = action.payload;
+                if (TableUtil.isFullyLoaded(tblId)) {
+                    TableStatsCntlr.dispatchLoadTblStats(TableUtil.findTblById(tblId).request);
+                }
+                break;
+            case TablesCntlr.TABLE_NEW_LOADED:
+                const {tbl_id} = action.payload;
+                tableStatsState = flux.getState()[TableStatsCntlr.TBLSTATS_DATA_KEY];
+                if (has(tableStatsState, tbl_id)) {
+                    TableStatsCntlr.dispatchLoadTblStats(request);
+                }
+                xyPlotState = flux.getState()[XYPlotCntlr.XYPLOT_DATA_KEY];
+                if (has(xyPlotState, tbl_id)) {
+                    const xyPlotParams = xyPlotState[tbl_id].xyPlotParams;
+                    XYPlotCntlr.dispatchLoadPlotData(xyPlotParams, request);
+                }
+                histogramState = flux.getState()[HistogramCntlr.HISTOGRAM_DATA_KEY];
+                if (has(histogramState, tbl_id)) {
+                    const histogramParams = xyPlotState[tbl_id].histogramParams;
+                    HistogramCntlr.dispatchLoadColData(histogramParams, request);
+                }
+                break;
+        }
+    }
+}
```

Comment:
'has' checks if there is statistics/xyplot/histogram connected to the
table. The load will be dispatched only if there is something connected.
But it does not take into account if the chart is displayed or not. I
should probably have some logic for lazy (delayed) load. If a chart is not
on display, the data should not load. The load should be dispatched only
when the chart becomes visible. I'll create a ticket for it.

2016-05-06 16:39 GMT-07:00 loitly <notifications@github.com>:

> In src/firefly/js/visualize/saga/ChartsSync.js
> <https://github.com/Caltech-IPAC/firefly/pull/79#discussion_r62404666>:
>
> > +                    TableStatsCntlr.dispatchLoadTblStats(request);
> > +                }
> > +                xyPlotState = flux.getState()[XYPlotCntlr.XYPLOT_DATA_KEY];
> > +                if (has(xyPlotState, tbl_id)) {
> > +                    const xyPlotParams = xyPlotState[tbl_id].xyPlotParams;
> > +                    XYPlotCntlr.dispatchLoadPlotData(xyPlotParams, request);
> > +                }
> > +                histogramState = flux.getState()[HistogramCntlr.HISTOGRAM_DATA_KEY];
> > +                if (has(histogramState, tbl_id)) {
> > +                    const histogramParams = xyPlotState[tbl_id].histogramParams;
> > +                    HistogramCntlr.dispatchLoadColData(histogramParams, request);
> > +                }
> > +                break;
> > +        }
> > +    }
> > +}
>
> Same question; if there is no histogram or xyplot, will it still dispatch
> the load?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Caltech-IPAC/firefly/pull/79/files/d21b15a078bcd06a2bc7634b6e1a98769057da2e#r62404666>
>